system/include/libc/bits/alltypes.h

```diff
@@ -88,7 +88,11 @@ typedef struct { union { int __i[9]; unsigned __s[9]; } __u; } pthread_attr_t;
 #endif
 
 #if defined(__NEED_pthread_mutex_t) && !defined(__DEFINED_pthread_mutex_t)
+#ifdef __EMSCRIPTEN__
+typedef struct { union { int __i[7]; void *__p[7]; } __u; } pthread_mutex_t;
```

Comment:
The musl algorithm for implementing a mutex differs in semantics for their `_m_lock` field, so I was not able to reuse that field while using the futex API that we have in the pthreads spec. Updated this with a comment. For a while I attempted to adapt our mutex implementation to reuse musl syscalls architecture, but that got hairy with the semaphores implementation that I concluded it's safer to use a separate field for the mutex lock control structure instead.